MapboxDirections/MBRouteOptions.swift

```diff
@@ -485,13 +485,15 @@ open class RouteOptions: NSObject, NSSecureCoding, NSCopying{
                 let coordinate = CLLocationCoordinate2D(geoJSON: location)
                 return Waypoint(coordinate: coordinate, name: waypoint.name ?? json["name"] as? String)
             }
+            self.waypoints = namedWaypoints
```

Comment:
I’m not sure that receiving a response should have the side effect of changing the original RouteOptions request. Consider that the developer might intentionally avoid naming one waypoint, but then it gets filled in with OSRM’s name for the waypoint. The next time the RouteOptions object gets reused, the name becomes explicit.

It seems like the key to this fix is the other change, which ensures that the returned array of waypoints is named.

Reply:
- [x] Don't update request model with response values.